Updates for both Jubrunr and Spring Boot 3 (#303)
AxonIQ/reference-guide: axon-framework/events/event-bus-and-event-store.md

#### `JpaEventStorageEngine`

> The `JpaEventStorageEngine` stores events in a JPA-compatible data source. The JPA event store stores events in entries. These entries contain the serialized form of an event, as well as some fields where metadata is stored for fast lookup of these entries. To use the `JpaEventStorageEngine`, you must have the JPA \(`jakarta.persistence`\) annotations on your classpath. The old, javax JPA \(`javax.persistence`\) annotations are also still supported. This will use an instance of the `JpaEventStorageEngine` using the `javax` namespace.

Reviewer: I may be wrong, but I think it should be this, right?

```suggestion
The `JpaEventStorageEngine` stores events in a JPA-compatible data source. The JPA event store stores events in entries. These entries contain the serialized form of an event, as well as some fields where metadata is stored for fast lookup of these entries. To use the `JpaEventStorageEngine`, you must have the JPA \(`jakarta.persistence`\) annotations on your classpath. The old, javax JPA \(`javax.persistence`\) annotations are also still supported. This will use an instance of the `JpaEventStorageEngine` using the `legacyjpa` namespace.
```

By default, the `JpaEventStorageEngine` requires an `EntityManagerProvider` implementation that returns the `EntityManager` instance for the `EventStorageEngine` to use. This also allows application managed persistence contexts to be used. It is the `EntityManagerProvider`'s responsibility to provide a correct instance of the `EntityManager`.

> There are a few implementations of the `EntityManagerProvider` available, each for different needs. The `SimpleEntityManagerProvider` simply returns the `EntityManager` instance which is given to it at construction time. This makes the implementation a simple option for container managed contexts. Alternatively, there is the `ContainerManagedEntityManagerProvider`, which returns the default persistence context, and is used by default by the JPA event store. Please note these classes are duplicated to accommodate for the `javax` to `jakarta` namespace change.

Reviewer: After seeing both changes, I think it's nicer to simply add a notice (similar to `> **Unique Key Constraint Consideration**`) about the `Jakarta or Javax?` question. That way, the pointers aren't spread in two spots, but combined in one. Furthermore, that means all documentation can focus on the Jakarta/current version instead, as this is where we want to steer the users to too.

Author: Not sure the update is what you had in mind, please check.

There are a few implementations of the `EntityManagerProvider` available, each for different needs. The `SimpleEntityManagerProvider` simply returns the `EntityManager` instance which is given to it at construction time. This makes the implementation a simple option for container managed contexts. Alternatively, there is the `ContainerManagedEntityManagerProvider`, which returns the default persistence context, and is used by default by the JPA event store.

> > **Move from javax.persistance to jkarta.persistance**

Reviewer: ```suggestion
> **Move from javax.persistance to jakarta.persistance**
```

The advantage of storing an entire commit in a single document is that commit is stored atomically. Furthermore, it requires only a single roundtrip for any number of events. The disadvantage is that it becomes harder to query events directly in the database. For example, when refactoring the domain model it is harder to "transfer" events from one aggregate to another if they are included in a "commit document".

> The `MongoEventStorageEngine` does not require a lot of configuration. All it needs is a reference to the collections to store the events in, and you're set to go. For production environments, you may want to double check the indexes on your collections. If you want transactions to be handled correctly, it's important to set a `TransactionManager`. Please not there are several other optional configuration properties, like the serializers to use and an optional upcaster chain.

Reviewer: ```suggestion
The `MongoEventStorageEngine` does not require a lot of configuration. All it needs is a reference to the collections to store the events in, and you're set to go. For production environments, you may want to double check the indexes on your collections. If you want transactions to be handled correctly, it's important to set a `TransactionManager`. Please note that there are several other optional configuration properties, like the serializers and an (optional) upcaster chain.
```
